src/libprojectM/Renderer/TextureManager.cpp

```diff
@@ -23,253 +22,375 @@
 #include "TextureManager.hpp"
 #include "Common.hpp"
 #include "IdleTextures.hpp"
+#include "Texture.hpp"
+#include "PerlinNoise.hpp"
+
 
+#define NUM_BLUR_TEX    6
 
 
-TextureManager::TextureManager(const std::string _presetURL): presetURL(_presetURL)
+TextureManager::TextureManager(const std::string _presetsURL, const int texsizeX, const int texsizeY):
+    presetsURL(_presetsURL)
 {
-#ifdef USE_DEVIL
-ilInit();
-iluInit();
-ilutInit();
-ilutRenderer(ILUT_OPENGL);
-#endif
+    extensions.push_back(".jpg");
+    extensions.push_back(".dds");
+    extensions.push_back(".png");
+    extensions.push_back(".tga");
+    extensions.push_back(".bmp");
+    extensions.push_back(".dib");
+
+    Preload();
+    loadTextureDir();
+
+    // Create main texture ans associated samplers
+    mainTexture = new Texture(texsizeX, texsizeY, false);
+    mainTexture->getSampler(GL_REPEAT, GL_LINEAR);
+    mainTexture->getSampler(GL_REPEAT, GL_NEAREST);
+    mainTexture->getSampler(GL_CLAMP_TO_EDGE, GL_LINEAR);
+    mainTexture->getSampler(GL_CLAMP_TO_EDGE, GL_NEAREST);
+    textures["main"] = mainTexture;
+
+    // Initialize blur textures
+    int w = texsizeX;
+    int h = texsizeY;
+    for (int i=0; i<NUM_BLUR_TEX; i++)
+    {
+        // main VS = 1024
+        // blur0 = 512
+        // blur1 = 256  <-  user sees this as "blur1"
+        // blur2 = 128
+        // blur3 = 128  <-  user sees this as "blur2"
+        // blur4 =  64
+        // blur5 =  64  <-  user sees this as "blur3"
+        if (!(i&1) || (i<2))
+        {
+            w = std::max(16, w/2);
+            h = std::max(16, h/2);
+        }
+        int w2 = ((w+3)/16)*16;
+        int h2 = ((h+3)/4)*4;
+
+        Texture * textureBlur = new Texture(w2, h2, false);
+        textureBlur->getSampler(GL_CLAMP_TO_EDGE, GL_LINEAR);
+        std::string texname = "blur" + std::to_string(i/2+1) + ((i%2) ? "" : "doNOTuseME");
+        textures[texname] = textureBlur;
+        blurTextures.push_back(textureBlur);
+    }
 
- Preload();
- loadTextureDir();
+    PerlinNoise noise;
+
+    GLuint noise_texture_lq_lite;
+    glGenTextures(1, &noise_texture_lq_lite);
+    glBindTexture(GL_TEXTURE_2D, noise_texture_lq_lite);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, 32, 32, 0, GL_LUMINANCE, GL_FLOAT, noise.noise_lq_lite);
```

Comment:
Not defined for me

<img width="355" alt="screen shot 2018-07-23 at 12 58 27" src="https://user-images.githubusercontent.com/245131/43070308-2f90a66c-8e78-11e8-9343-2fde0bb6655f.png">